### PR TITLE
ref: Session replay performance for SwiftUI

### PR DESCRIPTION
**Pull request by brustolin**
## :scroll: Description

Improved Performance for SwiftUI session replay

## :green_heart: How did you test it?

Samples

## :pencil: Checklist

You have to check all boxes before merging:

- [x] I reviewed the submitted code.
- [x] I added tests to verify the changes.
- [x] No new PII added or SDK only sends newly added PII if `sendDefaultPII` is enabled.
- [x] I updated the docs if needed.
- [x] Review from the native team if needed.
- [x] No breaking change or entry added to the changelog.
- [x] No breaking change for hybrid SDKs or communicated to hybrid SDKs.

### CODE REVIEW

File: Sources/Swift/Tools/SentryViewPhotographer.swift [hunk 1]
```diff
@@ -45,18 +45,24 @@ class SentryViewPhotographer: NSObject, SentryViewScreenshotProvider {
...
                 
+                var latestRegion: RedactRegion?
                 for region in redact {
                     let rect = CGRect(origin: CGPoint.zero, size: region.size)
                     var transform = region.transform
                     let path = CGPath(rect: rect, transform: &transform)
                     
+                    defer { latestRegion = region }
+                    
+                    guard region != latestRegion && imageRect.contains(path.boundingBoxOfPath) else { continue }
```
Comment (philipphofmann) [inline Sources/Swift/Tools/SentryViewPhotographer.swift:64]: `m`: It seems like this speeds up the performance significantly. Maybe it's worth adding a comment for this, cause we don't have a test to validate that we do this.

File: Sources/Swift/Tools/UIRedactBuilder.swift [hunk 2]
```diff
@@ -39,6 +39,10 @@ struct RedactRegion {
         self.type = type
         self.color = color
     }
+    
+    static func == (lhs: RedactRegion, rhs: RedactRegion) -> Bool {
+        lhs.size == rhs.size && lhs.transform == rhs.transform && lhs.type == rhs.type
```
[the remark this answers is not in the document]
Comment (brustolin) [reply Sources/Swift/Tools/UIRedactBuilder.swift:44]: Yeah, we can do that.

File: Sources/Swift/Tools/SentryViewPhotographer.swift (hunk 1 above)
Comment (brustolin) [reply Sources/Swift/Tools/SentryViewPhotographer.swift:64]: We do have a test to validate this.
Its the only test added to this PR

**Review by philipphofmann — APPROVED** [review]
LGTM

**Pull request merged**